src/sdam/monitor.rs

```diff
@@ -101,25 +101,11 @@ impl Monitor {
             // We only go to sleep when using the polling protocol (i.e. server never returned a
             // topologyVersion) or when the most recent check failed.
             if self.topology_version.is_none() || !check_succeeded {
-                #[cfg(test)]
-                let min_frequency = self
-                    .client_options
-                    .test_options
-                    .as_ref()
-                    .and_then(|to| to.min_heartbeat_freq)
-                    .unwrap_or(MIN_HEARTBEAT_FREQUENCY);
-
-                #[cfg(not(test))]
-                let min_frequency = MIN_HEARTBEAT_FREQUENCY;
-
-                tokio::select! {
```

Comment:
I moved the minimum delay into `wait_for_check_request` to consolidate the checks for the server closing.